storage/store.go

```diff
@@ -1271,33 +1280,78 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 			RangeLookup: args.Method() == proto.InternalRangeLookup,
 		}
 		bArgs.Add(pushArgs)
+		pushReply := &proto.InternalPushTxnResponse{}
+
+		// Check the txn cache first.
+		if s.pushedTxnCache.processPushedTxn(pushArgs, pushReply) {
+
+			cachedPushTxns = append(cachedPushTxns, &cachedPushTxn{
+				key:   intent.Key,
+				reply: pushReply,
+			})
+
+		} else {
+			bArgs.Add(pushArgs)
+			needPushIntents = append(needPushIntents, intent)
+		}
+
 	}
 	b := &client.Batch{}
 	b.InternalAddCall(client.Call{Args: bArgs, Reply: bReply})
 
-	// Run all pushes in parallel.
-	if pushErr := s.kvDB.Run(b); pushErr != nil {
-		if log.V(1) {
-			log.Infoc(ctx, "push for write intent %s failed: %s", wiErr, pushErr)
-		}
+	if len(needPushIntents) != 0 {
+		b := &client.Batch{}
```

Comment:
This shadows the `b` in the outer scope (line 1299). The outer one is no longer needed, right?